doc/source/user_guide/io.rst

```diff
@@ -272,9 +272,6 @@ parse_dates : boolean or list of ints or names or list of lists or dict, default
 
   .. note::
      A fast-path exists for iso8601-formatted dates.
-infer_datetime_format : boolean, default ``False``
```

Comment:
Hi, we can't simply remove it. We want to add a deprecation warning before removing it when we enforce the actual deprecation

Reply:
Yea, I asked for it in the issue but no one replied

Reply:
@phofl should this be Ok https://github.com/pandas-dev/pandas/pull/50334/commits/71466b6deef4045deca51b07969a97f288496bd8

Reply:
Hey - this parameter is now a no-op, so I'd say that it's correct to remove it from the user guide

It still shows in the docstring as deprecated, but I don't think there's any need to teach it in the user guide

Thoughts @phofl ?

Reply:
I would prefer mentioning that it is a no op but to keep it in here with a deprecation warning anyway. This should increase visibility, which is helpful with deprecations ( I think). Also keeps us consistent with what we are doing for other parameters in read_csv

Reply:
Sure but does a no-op need this much visibility? I think it looks confusing to beginners to read the user guide and find "you can use this parameter to speed things up. Though actually, never mind, it's deprecated, don't use it"

Ok for docstring, it's just for the user guide that I'm questioning the practice

I'll check tomorrow what's been done for other deprecations anyway, and what other libraries do

And sorry @ShashwatAgrawal20 for conflicting instructions here ☺